src/script/lua_api/l_object.cpp

```diff
@@ -834,6 +837,85 @@ int ObjectRef::l_get_properties(lua_State *L)
 	return 1;
 }
 
+// set_observers(self, observers)
+int ObjectRef::l_set_observers(lua_State *L)
+{
+	GET_ENV_PTR;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == nullptr)
+		throw LuaError("Invalid ObjectRef");
+
+	// Reset object to "unmanaged" (sent to everyone)?
+	if (lua_isnoneornil(L, 2)) {
+		sao->m_observers.reset();
+		return 0;
+	}
+
+	std::unordered_set<std::string> observer_names;
+	lua_pushnil(L);
+	while (lua_next(L, 2) != 0) {
+		std::string name = readParam<std::string>(L, -2);
+		if (name.empty())
+			throw LuaError("Observer name is empty");
+		if (name.size() > PLAYERNAME_SIZE)
+			throw LuaError("Observer name is too long");
+		if (!string_allowed(name, PLAYERNAME_ALLOWED_CHARS))
+			throw LuaError("Observer name contains invalid characters");
```

Comment:
Remark: (for the 3 conditions)
This might lead modders into manually sanitizing inputs (e.g. from formspecs, chat) for player names to fit exactly the limits currently defined within Minetest. However, they could avoid that by wrapping this function call inside a `(x)pcall`.

PS: Examples from other functions: (player-related functions during runtime)
* `AUTH.set_privileges(name, privs)` silently accepts invalid privileges and adds them regardless. Mods should do validity checks.
* `core.chat_send_player(name, msg)` ignored when no matching playername. No validity check needed.

Reply:
Modders should definitely validate *before* they feed player names into functions that accept them. Feeding unfiltered inputs straight to engine API functions is usually a very bad idea both in terms of user experience and security (for example players could try to waste server memory this way).

Reply:
The utility function addresses most of my concerns. Let's hope those separate conditions never go out of sync. Checking `minetest.player_exists` might be too expensive to run on every passed player name.